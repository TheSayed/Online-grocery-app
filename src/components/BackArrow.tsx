import React from "react";
import { TouchableOpacity, Image, StyleSheet } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { icons } from "../../assets/icons";
import { colors } from "../constants/colors";
import { scale, verticalScale } from "../utilis/scaling";

const BackArrow: React.FC = () => {
  const navigation = useNavigation();
  return (
    <TouchableOpacity
      onPress={() => {
        navigation.goBack();
      }}
      hitSlop={{ bottom: 30, left: 30, right: 30, top: 30 }}
      testID="back-arrow-button"
    >
      <Image
        source={icons.backArrow}
        style={styles.icon}
        testID="back-arrow-icon"
      />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  icon: {
    width: scale(9),
    height: verticalScale(14),
    tintColor: colors.primary,
  },
});

export default BackArrow;
