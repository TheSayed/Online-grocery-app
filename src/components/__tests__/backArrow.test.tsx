import React from "react";
import renderer from "react-test-renderer";
import { useNavigation } from "@react-navigation/native";
import BackArrow from "../BackArrow";

jest.mock("@react-navigation/native", () => ({
  useNavigation: jest.fn(),
}));

describe("<BackArrow />", () => {
  it("should match the snapshot", () => {
    const tree = renderer.create(<BackArrow />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call navigation.goBack when pressed", () => {
    const goBack = jest.fn();
    (useNavigation as jest.Mock).mockReturnValue({ goBack });

    const tree = renderer.create(<BackArrow />).root;
    const button = tree.findByProps({ testID: "back-arrow-button" });
    button.props.onPress();

    expect(goBack).toHaveBeenCalled();
  });
});
