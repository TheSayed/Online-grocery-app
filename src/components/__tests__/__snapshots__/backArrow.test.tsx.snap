// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackArrow /> should match the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 30,
      "left": 30,
      "right": 30,
      "top": 30,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="back-arrow-button"
>
  <Image
    source={1}
    style={
      {
        "height": 23,
        "tintColor": "rgb(255, 94, 0)",
        "width": 18,
      }
    }
    testID="back-arrow-icon"
  />
</View>
`;
